library/std/canon.qs

```diff
@@ -476,4 +476,51 @@ namespace Microsoft.Quantum.Canon {
         }
     }
 
+    /// # Summary
+    /// Applies Quantum Fourier Transform (QFT) to a quantum register.
+    ///
+    /// # Description
+    /// Applies QFT to a little-endian register `qs` of length n
+    /// containing |x₁⟩⊗|x₂⟩⊗…⊗|xₙ⟩. The qs[0] contains the
+    /// least significant bit xₙ. The state of qs[0] becomes
+    /// (|0⟩+𝑒^(2π𝑖[0.xₙ])|1⟩)/sqrt(2) after the operation.
+    ///
+    /// # Input
+    /// ## qs
+    /// Quantum register in a little-endian format to which the QFT is applied.
+    ///
+    /// # Reference
+    ///  - [Quantum Fourier transform](https://en.wikipedia.org/wiki/Quantum_Fourier_transform)
+    operation ApplyQFT (qs : Qubit[]) : Unit is Adj + Ctl {
+        let length = Length(qs);
+        Fact(length >= 1, "ApplyQFT: Length(qs) must be at least 1.");
+        for i in length-1..-1..0 {
+            H(qs[i]);
+            for j in 0..i-1 {
+                Controlled R1Frac([qs[i]], (1, j+1, qs[i-j-1]));
+            }
+        }
+    }
+
+    /// # Summary
+    /// Applies Quantum Fourier Transform (QFT) to a quantum register
+    /// and reverses the result to match the big-endian format.
+    operation QFTLE (qs : Qubit[]) : Unit is Adj + Ctl {
```

Comment:
This seems odd... is it intentional that the comment calls out the operation "reverses the result to match the big-endian format" but the operation is called `QFTLE`? Is this what you meant about maintaining the name or is it just a bug in the comment?

Reply:
This is very odd indeed and I'm glad you've asked. I spent a lot of time trying to figure out how I should comment this...
First, the truth. I only want to match the legacy QDK function with the same name. It would be weird if the legacy QDK function and the modern QDK function with the same name produced different outputs.
The legacy QDK operation [QFTLE](https://github.com/microsoft/QuantumLibraries/blob/a943e7087a45bbfcac57aea916d34c619d69325e/Standard/src/Canon/CommonGates.qs#L158C8-L158C8), through the series of transformations, was calling [ApproximateQFT](https://github.com/microsoft/QuantumLibraries/blob/a943e7087a45bbfcac57aea916d34c619d69325e/Standard/src/Canon/QFT.qs#L36), which was BE to BE. At least this is what [this comment](https://github.com/microsoft/QuantumLibraries/blob/a943e7087a45bbfcac57aea916d34c619d69325e/Standard/src/Canon/QFT.qs#L51) was saying. As a result, the input passed to QFTLE was transformed into BE format and the output of ApproximateQFT was passed back as is. That's why I decided to replicate this comment.
Frankly, I don't see the meaning of LE or BE on the output of QFT - I wouldn't call some bit "least significant" of "most significant". It's the exponent that's different from bit to bit. So I added a comment to ApplyQFT explaining where the corresponding bit is. ApplyQFT doesn't do any swaps, therefore it's faster (and is often acceptable). But it doesn't match any of the legacy QDK functions. So I added the QFTLE with this comment.
Alternatively I can add a different comment just explaining in which order it returns bits. I should also note that couldn't find any tests for QFTLE in the legacy QDK, only QFTBE, and one doesn't call the other. So it may very well be some bug...

Reply:
We decided to remove QFTLE entirely.